File: web/js/router/storyListR.js
const express = require('express');
const router = express.Router();

const path = require('path');

const load_data = require('./loadDataR');

router.get('/list', (req, res) => {
    res.sendFile(path.join(__dirname, '../../public/html/storyList.html'));
});
router.get('/get_story_list', (req, res) => {
    const data = load_data.load_all_novel_data();
    res.json(data);
})

module.exports = router;

File: web/python/router/storyListR.py
from fastapi import APIRouter, Request
from fastapi.templating import Jinja2Templates
import os

from func import loadStory

router = APIRouter(
    prefix="/list"
)

# 템플릿 경로 설정
template = Jinja2Templates(directory=os.path.join(os.path.dirname(__file__), "../../html"))

@router.get('/')
async def novel_list(request: Request):
    return template.TemplateResponse("storyList.html", {"request": request})

@router.get('/get_story_list')
async def get_story(request: Request):
    data = loadStory.load_all_novel_data()
    return {"data": data}
